README.md

```diff
@@ -182,112 +63,58 @@ take effect.
 
 ```
-$ git clone https://github.com/LedgerHQ/nanos-secure-sdk
-$ wget -O clang.tar.xz http://releases.llvm.org/4.0.0/clang+llvm-4.0.0-x86_64-linux-gnu-ubuntu-16.10.tar.xz
-$ wget -O gcc.tar.bz2 https://launchpadlibrarian.net/251687888/gcc-arm-none-eabi-5_3-2016q1-20160330-linux.tar.bz2
+docker run --rm -ti -v $(pwd):/app ghcr.io/ledgerhq/ledger-app-builder/ledger-app-dev-tools:latest
 ```
-
-Unzip the compilers and move them to appropriately-named directories:
+Then build the baking app inside the docker container shell as follows :
 
 ```
-$ mkdir bolos_env
-$ tar -xJf clang.tar.xz --directory bolos_env
-$ mv bolos_env/clang+llvm-4.0.0-x86_64-linux-gnu-ubuntu-16.10 bolos_env/clang-arm-fropi
-$ tar -xjf gcc.tar.bz2 --directory bolos_env
-```
-
-Set environment variables:
-
-```
-$ export BOLOS_SDK=$PWD/nanos-secure-sdk
-$ export BOLOS_ENV=$PWD/bolos_env
+BOLOS_SDK=$NANOS_SDK make
 ```
-
-To build the Tezos Wallet app:
-
+To enable debugging and log output, use
 ```
-$ APP=tezos_wallet make
-$ mv bin/app.hex wallet.hex
+BOLOS_SDK=$NANOS_SDK make DEBUG=1
 ```
-If this results in an error message that includes this line (possibly repeatedly):
+You can replace `NANOS` with `NANOSP`, `NANOX`, `STAX` for the other devices in BOLOS_SDK environmental variable.
 
+### Testing
+To test the application you need to enable python virtual environment and some dependencies. On any operating system, create a python virtual environment and activate it.
 ```
-#include <bits/libc-header-start.h>
+$ sudo apt-get update && sudo apt-get install -y qemu-user-static tesseract-ocr libtesseract-dev python3-pip libgmp-dev libsodium-dev git
+$ python3 -m venv env
+$ source env/bin/activate
 ```
-you may need to run:
-
+Inside the virtualenv, load the requirements.txt file.
 ```
-$ sudo apt-get install libc6-dev gcc-multilib g++-multilib
+(env)$ pip install -r test/python/requirements.txt
 ```
-and then re-run the `make` command.
-
-Note that if you build *both* apps, you need to run `make clean` before building
-the second one. So, to build both apps run:
-
+Now you can run ragger tests for any perticular ledger device. Please make sure you have built the app.elf files for that perticular device first. Then run following command:
 ```
-$ APP=tezos_wallet make
-$ mv bin/app.hex wallet.hex
-$ make clean
-$ APP=tezos_baking make
-$ mv bin/app.hex baking.hex
+(env)$ pytest test/python --device nanosp
 ```
+Replace nanosp with any of the following for respective device: nanos, nanosp, nanox , stax.
 
-To build just the Tezos Baking App:
-
-```
-$ APP=tezos_baking make
-$ mv bin/app.hex baking.hex
-```
 
 ### Installing the apps onto your Ledger device without Ledger Live
 
-Manually installing the apps requires a command-line tool called the
-[BOLOS Python Loader](https://ledger.readthedocs.io/projects/blue-loader-python/en/0.1.16/index.html).
+Manually installing the apps requires a command-line tool called LedgerBlue
```

Comment:
Is this section is equivalent to the `make load` command:
```
docker run --rm -ti -v "$(realpath .):/app" --privileged ghcr.io/ledgerhq/ledger-app-builder/ledger-app-dev-tools:latest bash -c "make load"
```

Reply:
Yes it is similar, recent docker image has some problems with udev rules and make load fails with OSError. many users reported this error on discord. Directly using the command from virtual environment works everywhere. So i included this command.